test/reductions/thomasvandoren/test/TestScanCounts.future

```diff
@@ -0,0 +1,3 @@
+bug: scan constructor fails to resolve
+
+See the error message in the .bad file.
```

Comment:
I think you are right about the inheritance/generic issue.

The generate method for scan is different in the paper and in my implementation. I implemented a version that works (see new commits). I also implemented a future-ized version that inherits the 'counts' class.